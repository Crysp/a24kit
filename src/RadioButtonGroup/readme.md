```js
class RadioButtonGroupExample extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            selected: '1'
        };
    }
    onCheck(selected) {
        this.setState({ selected });
    };
    render() {
        const { selected } = this.state;
        return (
            <RadioButtonGroup
                options={[
                    { value: "1", label: "Пункт 1" },
                    { value: "2", label: "Пункт 2" },
                    { value: "3", label: "Пункт 3" },
                    { value: "4", label: "Пункт 4" },
                    { value: "5", label: "Пункт 5" }
                ]}
                selected={selected}
                onCheck={this.onCheck.bind(this)}
            />
        );
    }
}

<RadioButtonGroupExample/>
```