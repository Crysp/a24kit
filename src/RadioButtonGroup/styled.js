import styled from 'styled-components';

export const List = styled.div`
    position: relative;
`;

export const Option = styled.div`
    position: relative;
    & + & {
        margin-top: 12px;
    }
`;
