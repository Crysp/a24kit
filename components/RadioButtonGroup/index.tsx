import React from 'react';
import RadioButton from '../RadioButton';
import { List, Option } from './styled';

export interface Props {
    selected: string
    options: {
        value: string
        label: string
    }[]
    color: 'blue' | 'green'
    large: boolean
    disabled: boolean
    onCheck(value: string): void
    'data-testid': string
}

export default class RadioButtonGroup extends React.Component<Props> {
    static defaultProps = {
        selected: null,
        color: 'blue',
        large: false,
        disabled: false,
        onCheck() {},
        'data-testid': undefined,
    };

    onCheck = (value: string) => {
        this.props.onCheck(value);
    };

    render() {
        const {
            selected,
            options,
            color,
            large,
            disabled,
            'data-testid': dataTestId,
        } = this.props;
        return (
            <List>
                {options.map(({ value, label }) => (
                    <Option key={value}>
                        <RadioButton
                            value={value}
                            label={label}
                            isChecked={selected === value}
                            onCheck={() => this.onCheck(value)}
                            color={color}
                            large={large}
                            disabled={disabled}
                            data-testid={dataTestId}
                        />
                    </Option>
                ))}
            </List>
        );
    }
}
