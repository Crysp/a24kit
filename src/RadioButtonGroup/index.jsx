import React from 'react';
import PropTypes from 'prop-types';
import RadioButton from '../RadioButton';
import { List, Option } from './styled';


/**
 * @develop
 */
export default class RadioButtonGroup extends React.Component {
    static propTypes = {
        /**
         * Выбранный элемент
         */
        selected: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
        /**
         * Набор данных для построения списка
         */
        options: PropTypes.arrayOf(PropTypes.shape({
            value: PropTypes.any,
            label: PropTypes.string,
        })).isRequired,
        /**
         * Цвет радио кнопок
         */
        color: PropTypes.oneOf(['blue', 'green']),
        /**
         * Большие радио кнопки
         */
        large: PropTypes.bool,
        /**
         * Неактивный список
         */
        disabled: PropTypes.bool,
        /**
         * Обработчик выбора элемента
         * @param value - выбранный элемент
         */
        onCheck: PropTypes.func,
        'data-testid': PropTypes.string,
    };

    static defaultProps = {
        selected: null,
        color: 'blue',
        large: false,
        disabled: false,
        onCheck() {},
        'data-testid': undefined,
    };

    onCheck = (value) => {
        this.props.onCheck(value);
    };

    render() {
        const {
            selected,
            options,
            color,
            large,
            disabled,
            'data-testid': dataTestId,
        } = this.props;
        return (
            <List>
                {options.map(({ value, label }) => (
                    <Option key={value}>
                        <RadioButton
                            value={value}
                            label={label}
                            isChecked={selected === value}
                            onCheck={() => this.onCheck(value)}
                            color={color}
                            large={large}
                            disabled={disabled}
                            data-testid={dataTestId}
                        />
                    </Option>
                ))}
            </List>
        );
    }
}
