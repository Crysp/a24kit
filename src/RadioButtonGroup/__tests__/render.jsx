import React from 'react';
import { create } from 'react-test-renderer';
import RadioButtonGroup from '../index';


const optionsMock = [
    { value: '1', label: 'Option 1' },
    { value: '2', label: 'Option 2' },
    { value: '3', label: 'Option 3' },
];


describe('Верстка компонента <RadioButtonGroup>', () => {
    it('Дефолтная', () => {
        const tree = create(<RadioButtonGroup options={optionsMock} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('С выбранным элементом', () => {
        const tree = create(
            <RadioButtonGroup options={optionsMock} selected={optionsMock[0].value} />,
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('Синего цвета', () => {
        const tree = create(<RadioButtonGroup options={optionsMock} color="blue" />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('Зеленого цвета', () => {
        const tree = create(<RadioButtonGroup options={optionsMock} color="green" />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('Увеличенного размера', () => {
        const tree = create(<RadioButtonGroup options={optionsMock} large />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('Неактивная', () => {
        const tree = create(<RadioButtonGroup options={optionsMock} disabled />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
