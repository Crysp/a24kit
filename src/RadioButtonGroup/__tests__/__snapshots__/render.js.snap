// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Верстка компонента <RadioButtonGroup> Дефолтная 1`] = `
<div
  className="sc-EHOje kNXPDk"
>
  <div
    className="sc-bZQynM kAwJPg"
  >
    <div
      className="sc-ifAKCX gPrOSU"
      disabled={false}
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="sc-bdVaJa cjORyY"
      >
        <div
          checked={false}
          className="sc-bwzfXH kbJlne"
          color="blue"
          disabled={false}
        />
        <div
          checked={false}
          className="sc-htpNat hQFrKh"
          color="blue"
          disabled={false}
        />
      </div>
      <div
        className="sc-bxivhb izKojs"
        disabled={false}
      >
        Option 1
      </div>
    </div>
  </div>
  <div
    className="sc-bZQynM kAwJPg"
  >
    <div
      className="sc-ifAKCX gPrOSU"
      disabled={false}
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="sc-bdVaJa cjORyY"
      >
        <div
          checked={false}
          className="sc-bwzfXH kbJlne"
          color="blue"
          disabled={false}
        />
        <div
          checked={false}
          className="sc-htpNat hQFrKh"
          color="blue"
          disabled={false}
        />
      </div>
      <div
        className="sc-bxivhb izKojs"
        disabled={false}
      >
        Option 2
      </div>
    </div>
  </div>
  <div
    className="sc-bZQynM kAwJPg"
  >
    <div
      className="sc-ifAKCX gPrOSU"
      disabled={false}
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="sc-bdVaJa cjORyY"
      >
        <div
          checked={false}
          className="sc-bwzfXH kbJlne"
          color="blue"
          disabled={false}
        />
        <div
          checked={false}
          className="sc-htpNat hQFrKh"
          color="blue"
          disabled={false}
        />
      </div>
      <div
        className="sc-bxivhb izKojs"
        disabled={false}
      >
        Option 3
      </div>
    </div>
  </div>
</div>
`;

exports[`Верстка компонента <RadioButtonGroup> Зеленого цвета 1`] = `
<div
  className="sc-EHOje kNXPDk"
>
  <div
    className="sc-bZQynM kAwJPg"
  >
    <div
      className="sc-ifAKCX gPrOSU"
      disabled={false}
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="sc-bdVaJa cjORyY"
      >
        <div
          checked={false}
          className="sc-bwzfXH kbJlne"
          color="green"
          disabled={false}
        />
        <div
          checked={false}
          className="sc-htpNat hEyLbp"
          color="green"
          disabled={false}
        />
      </div>
      <div
        className="sc-bxivhb izKojs"
        disabled={false}
      >
        Option 1
      </div>
    </div>
  </div>
  <div
    className="sc-bZQynM kAwJPg"
  >
    <div
      className="sc-ifAKCX gPrOSU"
      disabled={false}
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="sc-bdVaJa cjORyY"
      >
        <div
          checked={false}
          className="sc-bwzfXH kbJlne"
          color="green"
          disabled={false}
        />
        <div
          checked={false}
          className="sc-htpNat hEyLbp"
          color="green"
          disabled={false}
        />
      </div>
      <div
        className="sc-bxivhb izKojs"
        disabled={false}
      >
        Option 2
      </div>
    </div>
  </div>
  <div
    className="sc-bZQynM kAwJPg"
  >
    <div
      className="sc-ifAKCX gPrOSU"
      disabled={false}
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="sc-bdVaJa cjORyY"
      >
        <div
          checked={false}
          className="sc-bwzfXH kbJlne"
          color="green"
          disabled={false}
        />
        <div
          checked={false}
          className="sc-htpNat hEyLbp"
          color="green"
          disabled={false}
        />
      </div>
      <div
        className="sc-bxivhb izKojs"
        disabled={false}
      >
        Option 3
      </div>
    </div>
  </div>
</div>
`;

exports[`Верстка компонента <RadioButtonGroup> Неактивная 1`] = `
<div
  className="sc-EHOje kNXPDk"
>
  <div
    className="sc-bZQynM kAwJPg"
  >
    <div
      className="sc-ifAKCX eeEfuI"
      disabled={true}
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="sc-bdVaJa cjORyY"
      >
        <div
          checked={false}
          className="sc-bwzfXH iKRUrJ"
          color="blue"
          disabled={true}
        />
        <div
          checked={false}
          className="sc-htpNat iIsfHI"
          color="blue"
          disabled={true}
        />
      </div>
      <div
        className="sc-bxivhb kojMoR"
        disabled={true}
      >
        Option 1
      </div>
    </div>
  </div>
  <div
    className="sc-bZQynM kAwJPg"
  >
    <div
      className="sc-ifAKCX eeEfuI"
      disabled={true}
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="sc-bdVaJa cjORyY"
      >
        <div
          checked={false}
          className="sc-bwzfXH iKRUrJ"
          color="blue"
          disabled={true}
        />
        <div
          checked={false}
          className="sc-htpNat iIsfHI"
          color="blue"
          disabled={true}
        />
      </div>
      <div
        className="sc-bxivhb kojMoR"
        disabled={true}
      >
        Option 2
      </div>
    </div>
  </div>
  <div
    className="sc-bZQynM kAwJPg"
  >
    <div
      className="sc-ifAKCX eeEfuI"
      disabled={true}
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="sc-bdVaJa cjORyY"
      >
        <div
          checked={false}
          className="sc-bwzfXH iKRUrJ"
          color="blue"
          disabled={true}
        />
        <div
          checked={false}
          className="sc-htpNat iIsfHI"
          color="blue"
          disabled={true}
        />
      </div>
      <div
        className="sc-bxivhb kojMoR"
        disabled={true}
      >
        Option 3
      </div>
    </div>
  </div>
</div>
`;

exports[`Верстка компонента <RadioButtonGroup> С выбранным элементом 1`] = `
<div
  className="sc-EHOje kNXPDk"
>
  <div
    className="sc-bZQynM kAwJPg"
  >
    <div
      className="sc-ifAKCX gPrOSU"
      disabled={false}
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="sc-bdVaJa cjORyY"
      >
        <div
          checked={true}
          className="sc-bwzfXH bqyPxt"
          color="blue"
          disabled={false}
        />
        <div
          checked={true}
          className="sc-htpNat gWVkrB"
          color="blue"
          disabled={false}
        />
      </div>
      <div
        className="sc-bxivhb izKojs"
        disabled={false}
      >
        Option 1
      </div>
    </div>
  </div>
  <div
    className="sc-bZQynM kAwJPg"
  >
    <div
      className="sc-ifAKCX gPrOSU"
      disabled={false}
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="sc-bdVaJa cjORyY"
      >
        <div
          checked={false}
          className="sc-bwzfXH kbJlne"
          color="blue"
          disabled={false}
        />
        <div
          checked={false}
          className="sc-htpNat hQFrKh"
          color="blue"
          disabled={false}
        />
      </div>
      <div
        className="sc-bxivhb izKojs"
        disabled={false}
      >
        Option 2
      </div>
    </div>
  </div>
  <div
    className="sc-bZQynM kAwJPg"
  >
    <div
      className="sc-ifAKCX gPrOSU"
      disabled={false}
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="sc-bdVaJa cjORyY"
      >
        <div
          checked={false}
          className="sc-bwzfXH kbJlne"
          color="blue"
          disabled={false}
        />
        <div
          checked={false}
          className="sc-htpNat hQFrKh"
          color="blue"
          disabled={false}
        />
      </div>
      <div
        className="sc-bxivhb izKojs"
        disabled={false}
      >
        Option 3
      </div>
    </div>
  </div>
</div>
`;

exports[`Верстка компонента <RadioButtonGroup> Синего цвета 1`] = `
<div
  className="sc-EHOje kNXPDk"
>
  <div
    className="sc-bZQynM kAwJPg"
  >
    <div
      className="sc-ifAKCX gPrOSU"
      disabled={false}
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="sc-bdVaJa cjORyY"
      >
        <div
          checked={false}
          className="sc-bwzfXH kbJlne"
          color="blue"
          disabled={false}
        />
        <div
          checked={false}
          className="sc-htpNat hQFrKh"
          color="blue"
          disabled={false}
        />
      </div>
      <div
        className="sc-bxivhb izKojs"
        disabled={false}
      >
        Option 1
      </div>
    </div>
  </div>
  <div
    className="sc-bZQynM kAwJPg"
  >
    <div
      className="sc-ifAKCX gPrOSU"
      disabled={false}
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="sc-bdVaJa cjORyY"
      >
        <div
          checked={false}
          className="sc-bwzfXH kbJlne"
          color="blue"
          disabled={false}
        />
        <div
          checked={false}
          className="sc-htpNat hQFrKh"
          color="blue"
          disabled={false}
        />
      </div>
      <div
        className="sc-bxivhb izKojs"
        disabled={false}
      >
        Option 2
      </div>
    </div>
  </div>
  <div
    className="sc-bZQynM kAwJPg"
  >
    <div
      className="sc-ifAKCX gPrOSU"
      disabled={false}
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="sc-bdVaJa cjORyY"
      >
        <div
          checked={false}
          className="sc-bwzfXH kbJlne"
          color="blue"
          disabled={false}
        />
        <div
          checked={false}
          className="sc-htpNat hQFrKh"
          color="blue"
          disabled={false}
        />
      </div>
      <div
        className="sc-bxivhb izKojs"
        disabled={false}
      >
        Option 3
      </div>
    </div>
  </div>
</div>
`;

exports[`Верстка компонента <RadioButtonGroup> Увеличенного размера 1`] = `
<div
  className="sc-EHOje kNXPDk"
>
  <div
    className="sc-bZQynM kAwJPg"
  >
    <div
      className="sc-ifAKCX gPrOSU"
      disabled={false}
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="sc-bdVaJa ddwLo"
      >
        <div
          checked={false}
          className="sc-bwzfXH kbJlne"
          color="blue"
          disabled={false}
        />
        <div
          checked={false}
          className="sc-htpNat gtFvXH"
          color="blue"
          disabled={false}
        />
      </div>
      <div
        className="sc-bxivhb izKojs"
        disabled={false}
      >
        Option 1
      </div>
    </div>
  </div>
  <div
    className="sc-bZQynM kAwJPg"
  >
    <div
      className="sc-ifAKCX gPrOSU"
      disabled={false}
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="sc-bdVaJa ddwLo"
      >
        <div
          checked={false}
          className="sc-bwzfXH kbJlne"
          color="blue"
          disabled={false}
        />
        <div
          checked={false}
          className="sc-htpNat gtFvXH"
          color="blue"
          disabled={false}
        />
      </div>
      <div
        className="sc-bxivhb izKojs"
        disabled={false}
      >
        Option 2
      </div>
    </div>
  </div>
  <div
    className="sc-bZQynM kAwJPg"
  >
    <div
      className="sc-ifAKCX gPrOSU"
      disabled={false}
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="sc-bdVaJa ddwLo"
      >
        <div
          checked={false}
          className="sc-bwzfXH kbJlne"
          color="blue"
          disabled={false}
        />
        <div
          checked={false}
          className="sc-htpNat gtFvXH"
          color="blue"
          disabled={false}
        />
      </div>
      <div
        className="sc-bxivhb izKojs"
        disabled={false}
      >
        Option 3
      </div>
    </div>
  </div>
</div>
`;
